File: api/src/main/java/com/aa/weatherprediction/model/Report.java
package com.aa.weatherprediction.model;

import com.squareup.moshi.JsonClass;

import java.util.List;

@JsonClass(generateAdapter = true)
public record Report(
        List<String> alerts,
        List<String> mains,
        Double temp_min,
        Double temp_max
) {
    public Report() {
        this(List.of(), List.of(), null, null);
    }
}

File: weather-prediction-api/src/main/kotlin/com/aa/weatherprediction/model/Report.kt
package com.aa.weatherprediction.model

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Report (
    var alerts: List<String> = listOf(),
    var mains: List<String>? = listOf(),
    var temp_min: Double? = null,
    var temp_max: Double? = null,
)
